contracts/service-registry/src/state.rs

```diff
@@ -134,6 +135,10 @@ pub enum AuthorizationState {
 pub const SERVICES: Map<&str, Service> = Map::new("services");
 // maps (service_name, chain_name, worker_address) -> ()
 pub const WORKERS_PER_CHAIN: Map<(&str, &ChainName, &Addr), ()> = Map::new("workers_per_chain");
+type ChainNames = HashSet<ChainName>;
+// maps (service_name, worker_address) -> List of ChainNames
+pub const CHAINS_PER_WORKER: Map<(&str, &Addr), ChainNames> = Map::new("chains_per_worker");
```

Comment:
since you are already adding a new alias, I think it would make sense to add aliases for `&str` and `&Addr` as well instead of using a comment